framework/application/application.cpp

```diff
@@ -54,11 +54,114 @@ GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+                         decode::FileProcessor* file_processor) :
+    name_(name),
+    file_processor_(file_processor), cli_wsi_extension_(cli_wsi_extension), running_(false), paused_(false),
+    pause_frame_(0)
+{
+    bool success = true;
+    if (cli_wsi_extension_.empty())
+    {
+        // NOOP : If no WSI context is selected on the command line, each WSI extension
+        //  present in the VkInstanceCreateInfo will be handled when vkCreateInstance()
+        //  is processed.
+    }
+    else
+#if defined(VK_USE_PLATFORM_WIN32_KHR)
+        if (cli_wsi_extension_ == VK_KHR_WIN32_SURFACE_EXTENSION_NAME)
+    {
+        InitializeWsiContext(VK_KHR_WIN32_SURFACE_EXTENSION_NAME);
+    }
+    else
+#endif
+#if defined(VK_USE_PLATFORM_WAYLAND_KHR)
+        if (cli_wsi_extension_ == VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME)
+    {
+        InitializeWsiContext(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME);
+    }
+    else
+#endif
+#if defined(VK_USE_PLATFORM_XCB_KHR)
+        if (cli_wsi_extension_ == VK_KHR_XCB_SURFACE_EXTENSION_NAME)
+    {
+        InitializeWsiContext(VK_KHR_XCB_SURFACE_EXTENSION_NAME);
+    }
+    else
+#endif
+#if defined(VK_USE_PLATFORM_XLIB_KHR)
+        if (cli_wsi_extension_ == VK_KHR_XLIB_SURFACE_EXTENSION_NAME)
+    {
+        InitializeWsiContext(VK_KHR_XLIB_SURFACE_EXTENSION_NAME);
+    }
+    else
+#endif
+#if defined(VK_USE_PLATFORM_HEADLESS)
+        if (cli_wsi_extension_ == VK_EXT_HEADLESS_SURFACE_EXTENSION_NAME)
+    {
+        InitializeWsiContext(VK_EXT_HEADLESS_SURFACE_EXTENSION_NAME);
+    }
+    else
+#endif
+    {
+        GFXRECON_WRITE_CONSOLE("Failed to initialize WSI context [%s].  WSI initializaton will attempt to fallback to "
+                               "a supported WSI extension",
+                               cli_wsi_extension_.c_str());
+    }
+}
+
 Application ::~Application() {}
 
+const WsiContext* Application::GetWsiContext(const std::string& wsi_extension, bool auto_select) const
+{
+    auto itr = wsi_contexts_.end();
+
+    // If auto_select is enabled and a WSI extension was selected on the CLI,
+    //  attempt to get that WSI context
+    if (auto_select && !cli_wsi_extension_.empty())
+    {
+        itr = wsi_contexts_.find(cli_wsi_extension_);
+    }
+
+    // If we don't have a valid WSI context after potential auto_select, fallback
+    //  to the current API call request
+    if (itr == wsi_contexts_.end())
+    {
+        itr = wsi_contexts_.find(wsi_extension);
+    }
+
+    // If auto_select is enabled and we still don't have a valid WSI context, loop
+    //  through loaded extensions and return the first one that _isn't_ the one
+    //  specified on the CLI or requested by the current API call
+    if (auto_select && itr == wsi_contexts_.end())
+    {
+        for (const auto& wsi_context_itr : wsi_contexts_)
+        {
+            if (wsi_context_itr.first != cli_wsi_extension_ && wsi_context_itr.first != wsi_extension &&
```

Comment:
Do the previous two if-statements guarantee that you'd never find `cli_wsi_extension_` or `wsi_extension` in this loop?  If so, I suggest dropping the if-statement.

Reply:
You're correct...will remove

Reply:
done